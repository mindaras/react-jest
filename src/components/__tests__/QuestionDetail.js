import React from "react";
import renderer from "react-test-renderer";
import { mapStateToProps, QuestionDetailDisplay } from "../QuestionDetail";

describe("The Question Detail Component", () => {
  describe("The Container Element", () => {
    describe("mapStateToProps", () => {
      it("should map the state to props correctly", () => {
        const sampleQuestion = {
          question_id: 42,
          body: "Some body"
        };
        const appState = {
          questions: [sampleQuestion]
        };
        const ownProps = {
          question_id: 42
        };
        const componentState = mapStateToProps(appState, ownProps);

        expect(componentState).toEqual(sampleQuestion);
      });
    });
  });
  describe("The display element", () => {
    it("should not regress", () => {
      const tree = renderer.create(
        <QuestionDetailDisplay
          title="Some title"
          body="Some body"
          answer_count={2}
          tags={["hitchhiking", "exercising"]}
        />
      );

      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
});
