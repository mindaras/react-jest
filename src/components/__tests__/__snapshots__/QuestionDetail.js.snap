// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Question Detail Component The display element should not regress 1`] = `
<div>
  <h3
    className="mb-2"
  >
    Some title
  </h3>
  <div>
    <div
      className="mb-3"
    >
      <div>
        <code>
          hitchhiking
        </code>
        <code>
          exercising
        </code>
      </div>
    </div>
    <div>
      <p>
        Some body
      </p>
    </div>
    <div>
      2
       Answers
    </div>
  </div>
</div>
`;
